src/test/java/org/jenkinsci/plugins/oic/plugintest/PluginTestHelper.java

```diff
@@ -0,0 +1,273 @@
+                authentication.getPrincipal(),
+                "Should be logged-in as " + PluginTestHelper.TEST_USER_USERNAME);
+        User user = toUser(authentication);
+        assertEquals(
+                PluginTestHelper.TEST_USER_FULL_NAME,
+                user.getFullName(),
+                "Full name should be " + PluginTestHelper.TEST_USER_FULL_NAME);
+        return user;
+    }
+
+    /**
+     * Gets the authentication object from the web client.
+     *
+     * @return the authentication object
+     */
+    public static Authentication getAuthentication(JenkinsRule.WebClient webClient) {
+        try {
+            return webClient.executeOnServer(Jenkins::getAuthentication2);
+        } catch (Exception e) {
+            // safely ignore all exceptions, the method never throws anything
+            return null;
+        }
+    }
+
+    public static @Nullable User toUser(Authentication authentication) {
+        return User.get(String.valueOf(authentication.getPrincipal()), false, Map.of());
+    }
+
+    public static void expire(JenkinsRule.WebClient webClient) throws Exception {
+        webClient.executeOnServer(() -> {
+            Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+            User user = User.get2(authentication);
+            OicCredentials credentials = user.getProperty(OicCredentials.class);
+
+            // setting currentTimestamp == 1 guarantees this will be an expired cred
+            user.addProperty(new OicCredentials(
+                    credentials.getAccessToken(),
+                    credentials.getIdToken(),
+                    credentials.getRefreshToken(),
+                    60L,
+                    1L,
+                    60L));
+            return null;
+        });
+    }
+
+    public static void withoutRefreshToken(Map<String, String> token) {
+        token.compute("refresh_token", (o, n) -> null);
+    }
+
+    public static HttpResponse<String> getPageWithGet(@CheckForNull JenkinsRule jenkinsRule, String url)
+            throws IOException, InterruptedException {
+        // fix up the url, if needed
+        if (url.startsWith("/")) {
+            url = url.substring(1);
+        }
+
+        HttpClient c = HttpClient.newBuilder()
+                .followRedirects(HttpClient.Redirect.NEVER)
+                .build();
+        return c.send(
+                HttpRequest.newBuilder(URI.create(jenkinsRule.getURL() + url))
+                        .GET()
+                        .build(),
+                HttpResponse.BodyHandlers.ofString());
+    }
+
+    /**
+     * performs a GET request using a basic authorization header
+     *
+     * @param user  - The user id
+     * @param token - the password api token to user
+     * @param url   - the url to request
+     * @return HttpResponse
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public static HttpResponse<String> getPageWithGet(
```

Comment:
`@CheckForNull JenkinsRule jenkinsRule` but then ` HttpRequest.newBuilder(URI.create(jenkinsRule.getURL() + url))` so this should be an spotbugs warning about NPE.  (not sure why its not, possibly as it is test code.

but if you have a `JenkinsRule` just use a `WebClient` and ``withBasicCredentials`? 

`org.htmlunit.Page.getWebResponse().getContentAsString()` ?